packages/utils/src/lib/git.integration.test.ts

```diff
@@ -60,4 +76,86 @@ describe('git utils', () => {
+  });
+
+  it('statusIsClean should return true if no changes are given', async () => {
+    await expect(statusIsClean(git)).resolves.toBe(true);
+  });
+
+  it('guardAgainstLocalChanges should throw if history is dirty', async () => {
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    await expect(guardAgainstLocalChanges(git)).rejects.toThrow(
+      'Working directory needs to be clean before we you can proceed. Commit your local changes or stash them.',
+    );
+  });
+
+  it('guardAgainstLocalChanges should not throw if history is clean', async () => {
+    await expect(guardAgainstLocalChanges(git)).resolves.toBeUndefined();
+  });
+
+  it('safeCheckout should checkout target branch in clean state', async () => {
+    await expect(git.branch()).resolves.toEqual(
+      expect.objectContaining({ current: 'master' }),
+    );
+    await expect(
+      safeCheckout('feature-branch', {}, git),
+    ).resolves.toBeUndefined();
+    await expect(git.branch()).resolves.toEqual(
+      expect.objectContaining({ current: 'feature-branch' }),
+    );
+  });
+
+  it('safeCheckout should throw if history is dirty', async () => {
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    await expect(safeCheckout('master', {}, git)).rejects.toThrow(
+      'Working directory needs to be clean before we you can proceed. Commit your local changes or stash them.',
+    );
+  });
+
+  it('safeCheckout should clean local changes and check out to master', async () => {
+    // needs to get reset to be clean
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    // needs to get cleaned to be clean
+    await writeFile(join(baseDir, 'README.md'), '# hello-world-2\n');
+
+    await expect(
+      safeCheckout('feature-branch', { forceCleanStatus: true }, git),
+    ).resolves.toBeUndefined();
+    await expect(git.branch()).resolves.toEqual(
+      expect.objectContaining({ current: 'feature-branch' }),
+    );
+  });
+
+  it('getCurrentBranchOrTag should log current branch', async () => {
+    await expect(getCurrentBranchOrTag(git)).resolves.toBe('master');
+  });
+});
+
+describe('git utils in a git repo without a branch and commits', () => {
+  const baseDir = join(process.cwd(), 'tmp', 'testing-git-repo');
+  let git: SimpleGit;
+
+  beforeAll(async () => {
+    await mkdir(baseDir, { recursive: true });
+    git = simpleGit(baseDir);
+    await git.init();
+  });
+
+  afterAll(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  it('getCurrentBranchOrTag should throw if no branch is given', async () => {
+    await expect(getCurrentBranchOrTag(git)).rejects.toThrow(
+      "git: 'describe --tags --exact-match' is not a git command. See 'git --help'",
```

Comment:
This command should return `fatal: No names found, cannot describe anything.` or similar. 

However, the asserted error message sounds like something went wrong with the command itself.

Reply:
I can add a custom error. What should be the message?
(A suggestion helps as my formulations of Error are improvable always, plus i don't know git that well to fully understand all situations in which this function could potentially throw)

Reply:
message is no "could not checkout current branch or tag"